forest/daemon/src/daemon.rs

```diff
@@ -377,8 +377,22 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
 
     services.spawn(p2p_service.run());
 
-    // Block until ctrl-c is hit
-    ctrlc_oneshot.await.unwrap();
+    loop {
+        select! {
+            option = services.join_next().fuse() => {
+                if let Some(res) = option {
+                    if let Ok(res_inner) = res {
+                        if let Err(error_message) = res_inner {
```

Comment:
if we're not handling other clauses for the above two matches, I think we we can do: 
```
if let Some(Ok(Err(error_message)) = option {

}
```